src/wallet/wallet.cpp

```diff
@@ -1050,6 +1050,7 @@ bool CWallet::AddToWalletIfInvolvingMe(const CTransactionRef& ptx, const CBlockI
                     std::map<CKeyID, int64_t>::const_iterator mi = m_pool_key_to_index.find(keyid);
                     if (mi != m_pool_key_to_index.end()) {
                         LogPrintf("%s: Detected a used keypool key, mark all keypool key up to this key as used\n", __func__);
+                        AddRelatedScripts(mi->first);
```

Comment:
I've renamed the CKeyStore::AddRelatedKeyScripts functions to CKeyStore::ImplicitlyLearnRelatedKeyScripts. It's not usable because that function only adds the script in memory. The point of AddRelatedScript is to make them explicit, and it will only be called on keys whose related scripts are already implicitly in the keystore.